packages/client/lib/sync/fullsync.ts

```diff
@@ -223,6 +176,43 @@ export class FullSynchronizer extends Synchronizer {
     })
   }
 
+  async processBlocks(execution: VMExecution, blocks: Block[] | BlockHeader[]) {
+    if (this.config.chainCommon.gteHardfork(Hardfork.Merge)) {
+      // If we are beyond the merge block we should stop the fetcher
+      this.config.logger.info('Merge hardfork reached, stopping block fetcher')
+      if (this.fetcher) {
+        this.fetcher.clear()
+        this.fetcher.destroy()
+        this.fetcher = null
+      }
+    }
+
+    if (blocks.length === 0) {
+      this.config.logger.warn('No blocks fetched are applicable for import')
+      return
+    }
+
+    blocks = blocks as Block[]
+    const first = new BN(blocks[0].header.number)
+    const hash = short(blocks[0].hash())
+    const baseFeeAdd = this.config.chainCommon.gteHardfork(Hardfork.London)
+      ? `baseFee=${blocks[0].header.baseFeePerGas} `
+      : ''
+    this.config.logger.info(
+      `Imported blocks count=${
+        blocks.length
+      } number=${first} hash=${hash} ${baseFeeAdd}hardfork=${this.config.chainCommon.hardfork()} peers=${
+        this.pool.size
+      }`
+    )
+    this.txPool.removeNewBlockTxs(blocks)
+
+    if (this.running || this.config.chainCommon.gteHardfork(Hardfork.Merge)) {
```

Comment:
I think we can remove this here now `this.config.chainCommon.gteHardfork(Hardfork.Merge)` since in engine.ts I manually do `execution.run()`. it was an artifact of having execution as part of sync, but now that we don't, it would no longer be needed

Reply:
I guess, since beacon sync/optimistic will have its own synchronizer, it makes sense :+1: